main.go

```diff
@@ -89,9 +141,14 @@ func main() {
 		os.Exit(1)
 	}
 
+	promErrCh := make(chan error)
+	go runPrometheusServer(metricsAddr, promErrCh)
+	monitor := newPromMonitor(promErrCh)
```

Comment:
Is the whole monitor required?
From what it does right now, I think you could also just log the error inside `runPrometheusServer` as nothing else happens when the server fails.

Reply:
You are right - this is useless. But it looked so smart and fancy :p